lib/streamlit/__init__.py

```diff
@@ -626,3 +627,74 @@ def _maybe_print_repl_warning():
                 ),
                 script_name,
             )
+
+
+# Feature namespaces
+
+class _BetaNamespace(object):
+    """This namespace holds Streamlit features that are in currently in beta.
+
+    "Beta features" are features that are planned to become part of the
+    Streamlit core but whose APIs/behavior have not stabilized yet. They may
+    change in ways that are not backward-compatible.
+
+    The lifecycle of a beta feature is:
+
+    1. Feature is added to the beta namespace.
+    2. After a few months: the feature's API stabilizes and the feature is
+       *cloned* into the `st` namespace, so it will exist in both `st` and
+       `st.beta`. At this point, users will see a warning when using the
+       version of the feature that lives in the beta namespace -- but the
+       `st.beta` feature will otherwise still work.
+    3. After 3 months: the feature is *removed* from the `st.beta` namespace,
+       but there will still be a stub in `st.beta` that shows an error with
+       appropriate instructions.
+    4. After 3 months: the stub in `st.beta` is removed.
+
+    The main difference between `st.beta` and `st.experimental` is that beta
+    features will always make it into the `st` namespace at some point, while
+    experimental ones often do not.
+
+    """
+    # Add beta features here. For example:
+    #foo = _foo
+    pass
+
+
+class _ExperimentalNamespace(object):
+    """This namespace holds highly experimental Streamlit features.
+
+    "Experiment features" are features that we are still trying to understand.
+    If they are successful, they will at some point become part of core
+    Streamlit, by moving to the `st.beta` namespace and then core `st`. If
+    unsuccessful, they may be removed.
+
+    Why this exists: we used to provide several Github Gists that allowed
+    people to hack experimental features into Streamlit, but is was hard to
```

Comment:
Typo: "but is was hard" -> "but it was hard"

Reply:
🦅